### ISSUE TITLE
I use Detr to train my data set and get map = 0

### ISSUE THREAD

**Issue by mstarget** [open-mmlab/mmdetection#4929]
I use Detr to train my data set and get map = 0

I use version 2.7 using Detr_ r50_ 8x2_ 150e_ Coco model training their own data sets (coco format), samples_ per_ The GPU parameter is set to 4, and the rest of the parameters are all by default. Why is the AP and AR on the verification set 0 after iterating to generation 70? How can I solve this problem

`
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=2.70s).
Accumulating evaluation results...
DONE (t=1.15s).
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = 0.000
Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = 0.000
Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.000
Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.000
Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = 0.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = 0.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.000

`

**Comment by ZwwWayne** [comment]
There are some hyper parameters like learning rate or some thing else need to be specifically tuned for your dataset. We have no idea about that.

**Comment by gingerbread8** [comment]
I once had this problem! AP and recall are close to 0, but after I modify the learning rate, recall is 0.33 and AP is 0.0015 in the eighth epoch. I don't know whether my parameters still need to be modified, or whether the convergence of the model is so slow。Have you solved the problem? We can talk about it

**Comment by mstarget** [comment]
&nbsp;I've tried different optimizers, I've tried different learning rates (from 0.1 to 0.0000001), but I've got 0 map




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "open-mmlab/mmdetection"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年4月16日(星期五) 下午4:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [open-mmlab/mmdetection] I use Detr to train my data set and get map = 0 (#4929)





 
I once had this problem! AP and recall are close to 0, but after I modify the learning rate, recall is 0.33 and AP is 0.0015 in the eighth epoch. I don't know whether my parameters still need to be modified, or whether the convergence of the model is so slow。Have you solved the problem? We can talk about it

**Comment by NeelBhowmik** [comment]
Hi,
I am facing a similar problem with DETR using a custom dataset (_custom_db_). The mAP is always 0. I tried different param configurations. 
Observations:
1. The loss is decreasing during training.
2. When I try to infer using DETR model weight (_custom_db_), it is predicting 100 bbox with a very low score. The 100 bbox coordinates and scores are exactly the same for different images.

Could you please take a look?

Thanks,
Neel

**Comment by gingerbread8** [comment]
solution1: There is a formula for adjusting the learning rate. lr = 0.00125*batch_size
solution2: You can load pretrained weight.I remembered this file is trained by cocoDataset . So when I used ,I find a code from internet to change the weight file to match my dataset
All my solution.
Good luck!


| |
zhouxiangjun5450
|
|
***@***.***
|
签名由网易邮箱大师定制
On 4/23/2021 ***@***.***> wrote：

&nbsp;I've tried different optimizers, I've tried different learning rates (from 0.1 to 0.0000001), but I've got 0 map




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "open-mmlab/mmdetection" ***@***.***&gt;;
发送时间:&nbsp;2021年4月16日(星期五) 下午4:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [open-mmlab/mmdetection] I use Detr to train my data set and get map = 0 (#4929)






I once had this problem! AP and recall are close to 0, but after I modify the learning rate, recall is 0.33 and AP is 0.0015 in the eighth epoch. I don't know whether my parameters still need to be modified, or whether the convergence of the model is so slow。Have you solved the problem? We can talk about it

**Comment by gingerbread8** [comment]
Sorry, I just saw the message.
My solution:
1. change learning rate. lr = batchsize * 0.00125
2.use pretrained weight. I remembered the weight was trained by cocodataSet. So when I use it , I find a code from internet to change the file  to match my dataSet
Hope my solution can give you  help


| |
zhouxiangjun5450
|
|
***@***.***
|
签名由网易邮箱大师定制
On 6/4/2021 22:50，Neelanjan ***@***.***> wrote：

Hi,
I am facing a similar problem with DETR using a custom dataset (custom_db). The mAP is always 0. I tried different param configurations.
Observations:

The loss is decreasing during training.
When I try to infer using DETR model weight (custom_db), it is predicting 100 bbox with a very low score. The 100 bbox coordinates and scores are exactly the same for different images.

Could you please take a look?

Thanks,
Neel
